JPN-554 Anon state for top contributors

perjg opened the pull request:
@Wikia/spitfires @d34th4ck3r

perjg committed:
JPN-554 Anon state for top contributors

Grunny commented on extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:
```diff
@@ -71,6 +71,13 @@ public function getTopContributorsData() {
+		$uselang  = 'uselang=' . $wgLang->getCode();
+		$login = '<a href="https://www.wikia.com/signin?' . $redirect . '&' . $uselang . '">' . $this->msg( 'communitypage-anon-login' )->escaped() . '</a>';
+		$register = '<a href="https://www.wikia.com/register?' . $redirect . '&' . $uselang . '">' . $this->msg( 'communitypage-anon-register' )->escaped() . '</a>';
+		$anonText = $this->msg( 'communitypage-anon-contrib-header', $login, $register )->plain();
```
The message should still be escaped, so if you need to pass raw HTML variables like this, use `rawParams`, i.e. `$this->msg( 'communitypage-anon-contrib-header' )->rawParams( $login, $register )->escaped();` This injects the parameters unescaped after the main message is safely escaped, but should of course only be used if all the raw parameters are known to be safe.

Grunny commented on extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:
```diff
@@ -71,6 +71,13 @@ public function getTopContributorsData() {
 		$topContributorsDetails = $this->getContributorsDetails( $topContributors );
 
+		global $wgLang;
+		$redirect = 'redirect=' . urlencode( SpecialPage::getTitleFor( 'CommunityPage' )->getCanonicalURL() );
```
Isn't the special page name just `Community`? Also, since you're within the special page class here, can't you just use `$this->getTitle()->getCanonicalURL()` since `$this->getTitle()` references the Community special page?

Grunny commented on extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:
```diff
@@ -71,6 +71,13 @@ public function getTopContributorsData() {
 
+		global $wgLang;
+		$redirect = 'redirect=' . urlencode( SpecialPage::getTitleFor( 'CommunityPage' )->getCanonicalURL() );
+		$uselang  = 'uselang=' . $wgLang->getCode();
```
You should be able to avoid the global with `$this->getLanguage()` instead of `$wgLang`.

Also, might be slightly cleaner to build the query string once since it's the same for both links, i.e.

``` php
$query = wfArrayToCGI( [
    'redirect' => $this->getTitle()->getCanonicalURL(),
    'uselang' => $this->getLanguage()->getCode(),
] );

$login = '<a href="https://www.wikia.com/signin?' . $query . '" // etc.
```

I'd also consider using the `Html` class (which handles escaping for you) to build a HTML string for the links rather than manual string concatenation, i.e.

``` php
$login = Html::element(
    'a',
    [ 'href' => 'https://www.wikia.com/signin?' . $query ],
    $this->msg( 'communitypage-anon-login' )->plain()
);
```

perjg committed:
JPN-554 Code cleanup

perjg committed:
JPN-554 Don't show ranking for anon users

perjg committed:
Merge branch 'dev' into JPN-554

perjg commented:
@Grunny Updated the code based on your comments, also added a commit to adjust the UI based on new discussion in the Jira task.

perjg committed:
JPN-554 Fixed indentation

Grunny commented on extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:
```diff
@@ -71,6 +71,25 @@ public function getTopContributorsData() {
+			$this->msg( 'communitypage-anon-register' )->plain()
+		);
+
+		$anonText = $this->msg( 'communitypage-anon-contrib-header', $login, $register )->plain();
```
Comment from before still applies. The message should be `->escaped()` and use `rawParams` to pass the link parameters instead.

perjg committed:
JPN-554 Correctly escape string

Grunny commented on extensions/wikia/CommunityPage/CommunityPage.i18n.php:
```diff
@@ -167,6 +170,10 @@
+	'communitypage-anon-contrib-header' => 'Text to show for anonymous users in the heading for top contributors. Tells users to login or register',
+	'communitypage-anon-login' => 'Log in text to use in the context of communitypage-anon-contrib-header $1 together with url',
+	'communitypage-anon-register' => 'register text to use in the context of communitypage-anon-contrib-header $1 together with url',
+
```
Unnecessary new line.

Grunny commented on extensions/wikia/CommunityPage/CommunityPage.i18n.php:
```diff
@@ -167,6 +170,10 @@
 	'communitypage-policy-module-link-page-name' => 'Name of the localized community policy page without Help namespace prefix.',
+	'communitypage-anon-contrib-header' => 'Text to show for anonymous users in the heading for top contributors. Tells users to login or register',
+	'communitypage-anon-login' => 'Log in text to use in the context of communitypage-anon-contrib-header $1 together with url',
+	'communitypage-anon-register' => 'register text to use in the context of communitypage-anon-contrib-header $1 together with url',
```
Should be `$2`?

Grunny commented:
Minor comments, otherwise LGTM.

perjg committed:
JPN-554 Fixed string in translation file. Removed anon text from modal

perjg commented:
@Grunny I fixed your minor comments for the i18n file.
I also removed one section from the top contributors module, as I had misunderstood the text in the Jira ticket, and it wasn't supposed to be there.

Grunny commented:
:+1:

perjg committed:
JPN-554 Downloaded latest translations from crowdin

perjg commented:
The new commit adds the latest translations from crowdin.
